docs/hardware/4-nrf91/2-quickstart/3-flash-firmware/2-bootloader.md

```diff
@@ -0,0 +1,123 @@
+
+1. Install the Go Language
+
+    :::note
+    You can install the Go language with `sudo apt install golang-go`, but some package managers have older versions. The following process will install the newest version.
+    :::
+
+    Visit [the Go language download page](https://go.dev/dl/) and download the latest version. Install the package:
+
+    ```bash
+    cd ~
+    tar -xvf go1.17.6.linux-amd64.tar.gz --directory ~/.
+    nano ~/.profile
+    ```
+
+    Add the following lines to the end of this file:
+
+    ```bash
+    export GOPATH=$HOME/go
+    export PATH=$PATH:$GOPATH/bin
+    ```
+
+    Save the file and load the changes:
+
+    ```bash
+    source ~/.profile
+    ```
+
+2. Install Newt Manager (newtmgr)
+
+    ```bash
+    mkdir -p $GOPATH/src/mynewt.apache.org/newtmgr
+    git clone https://github.com/circuitdojo/mynewt-newtmgr.git $GOPATH/src/mynewt.apache.org/newtmgr
+    cd $GOPATH/src/mynewt.apache.org/newtmgr/newtmgr
+    GO111MODULE=on go build
+    go install
+    ```
+
+## Flashing Firmware
+
+The nRF9160 can be programmed via USB using the MCUboot bootloader. The process depends on three steps: compiling your application for MCUboot, placing the target board in Device Firmware Upgrade (DFU) mode, and using a device management tool to write the firmware binary to the target.
+
+### Compile your code with MCUboot in mind
+
+1. Add the `CONFIG_BOOTLOADER_MCUBOOT` flag to the project's `prj.conf` file.
+
+    ```bash
+    CONFIG_BOOTLOADER_MCUBOOT=y
+    ```
+
+2. Build the project:
+
+    ```bash
+    cd ~/zephyr-nrf/modules/lib/golioth
+    #Flashing example for Circuit Dojo nRF9160 Feather
+    west build -b circuitdojo_feather_nrf9160_ns samples/basic/blinky -p
+    #Flashing example for Thiny:91
+    west build -b thingy91_nrf9160_ns samples/basic/blinky -p
+    ```
+
+3. When configured for MCUboot, the build process will compile an additional file for use with DFU. This file is located at:
+
+    ```bash
+    {project_dir}/build/zephyr/app-update.bin
+    ```
+
+### Flash with MCUboot
+
+1. Put the board into DFU mode by holding the bootloader button and power cycling the device.
+
+    * **Circuit Dojo nRF9160 Feather:** Hold down the user button and click the RST button. Keep holding the user button until the blue light comes on
+    * **Thingy:91:** Remove the orange case. Hold down the button in the middle of the device and cycle the on/off switch located next to the USB (this hardware lacks a reset button and there is an onboard battery)
+
+2. Use `newtmgr` to flash the firmware.
+
+    ```bash
+    #Flashing example for Circuit Dojo nRF9160 Feather
+    newtmgr --conntype=serial --connstring='dev=/dev/ttyUSB0,baud=1000000' image upload build/zephyr/app_update.bin
```

Comment:
Might need to add note about adding user to dialout

`sudo adduser <your-user> dialout`

Reply:
I added a section at the top of https://github.com/golioth/docs/pull/158/commits/f29c818c8b18d518915624383ebe5344242e7dfd# that guides user to add themselves to dialout.